crates/key-value-azure/src/lib.rs

```diff
@@ -0,0 +1,141 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use azure_data_cosmos::{
+    prelude::{AuthorizationToken, CollectionClient, CosmosClient, Query},
+    CosmosEntity,
+};
+use futures::StreamExt;
+use serde::{Deserialize, Serialize};
+use spin_core::async_trait;
+use spin_key_value::{log_error, Error, Store, StoreManager};
+
+pub struct KeyValueAzureCosmos {
+    client: CollectionClient,
+}
+
+impl KeyValueAzureCosmos {
+    pub fn new(key: String, account: String, database: String, container: String) -> Result<Self> {
+        let token = AuthorizationToken::primary_from_base64(&key).map_err(log_error)?;
+        let cosmos_client = CosmosClient::new(account, token);
+        let database_client = cosmos_client.database_client(database);
+        let client = database_client.collection_client(container);
+
+        Ok(Self { client })
+    }
+}
+
+#[async_trait]
+impl StoreManager for KeyValueAzureCosmos {
+    async fn get(&self, name: &str) -> Result<Arc<dyn Store>, Error> {
+        Ok(Arc::new(AzureCosmosStore {
+            _name: name.to_owned(),
+            client: self.client.clone(),
+        }))
+    }
+}
+
+struct AzureCosmosStore {
+    _name: String,
+    client: CollectionClient,
+}
+
+#[async_trait]
+impl Store for AzureCosmosStore {
+    async fn get(&self, key: &str) -> Result<Vec<u8>, Error> {
+        let pair = self.get_pair(key).await?;
+        Ok(pair.value)
+    }
+
+    async fn set(&self, key: &str, value: &[u8]) -> Result<(), Error> {
+        let pair = Pair {
+            id: key.to_string(),
+            value: value.to_vec(),
+        };
+        self.client
+            .create_document(pair)
+            .is_upsert(true)
+            .await
+            .map_err(log_error)?;
+        Ok(())
+    }
+
+    async fn delete(&self, key: &str) -> Result<(), Error> {
+        let pair = self.get_pair(key).await?;
```

Comment:
It would be better to use `self.exists(key).await?`

Reply:
Reading the interface, we should not return the error if the pair does not exist: https://github.com/fermyon/spin/blob/main/wit/preview2/key-value.wit#L56-L61

Reply:
Updated this to follow the spec and to use `self.exists`.